lib/modules/datasource/go/base.ts

```diff
@@ -53,120 +53,137 @@ export class BaseGoDatasource {
+              gitlabModuleName.indexOf('.git')
+            ),
           };
         }
-
         return {
           datasource: GitlabTagsDatasource.id,
           registryUrl: gitlabUrl,
-          packageName: gitlabUrlName,
+          packageName: gitlabModuleName,
         };
       }
 
-      const opts = hostRules.find({
-        hostType: PlatformId.Gitlab,
-        url: goSourceUrl,
-      });
-      if (opts.token) {
-        // get server base url from import url
-        const parsedUrl = URL.parse(goSourceUrl);
+      return {
+        datasource: GitlabTagsDatasource.id,
+        registryUrl: gitlabUrl,
+        packageName: gitlabUrlName,
+      };
+    }
 
-        // TODO: `parsedUrl.pathname` can be undefined
-        const packageName = trimLeadingSlash(`${parsedUrl.pathname}`);
+    const opts = hostRules.find({
+      hostType: PlatformId.Gitlab,
+      url: goSourceUrl,
+    });
+    if (opts.token) {
+      // get server base url from import url
+      const parsedUrl = URL.parse(goSourceUrl);
 
-        const registryUrl = `${parsedUrl.protocol}//${parsedUrl.host}`;
+      // TODO: `parsedUrl.pathname` can be undefined
+      const packageName = trimLeadingSlash(`${parsedUrl.pathname}`);
 
-        return {
-          datasource: GitlabTagsDatasource.id,
-          registryUrl,
-          packageName,
-        };
-      }
-    } else {
-      // GitHub Enterprise only returns a go-import meta
-      const importMatch = regEx(
-        `<meta\\s+name="go-import"\\s+content="([^\\s]+)\\s+([^\\s]+)\\s+([^\\s]+)">`
-      ).exec(res);
-      if (importMatch) {
-        const [, prefix, , goImportURL] = importMatch;
-        if (!goModule.startsWith(prefix)) {
-          logger.trace({ goModule }, 'go-import header prefix not match');
-          return null;
-        }
-        logger.debug({ goModule, goImportURL }, 'Go lookup import url');
-
-        // get server base url from import url
-        const parsedUrl = URL.parse(goImportURL);
+      const registryUrl = `${parsedUrl.protocol}//${parsedUrl.host}`;
 
-        // split the go module from the URL: host/go/module -> go/module
-        // TODO: `parsedUrl.pathname` can be undefined
-        const packageName = trimTrailingSlash(`${parsedUrl.pathname}`)
-          .replace(regEx(/\.git$/), '')
-          .split('/')
-          .slice(-2)
-          .join('/');
+      return {
+        datasource: GitlabTagsDatasource.id,
+        registryUrl,
+        packageName,
+      };
+    }
+    /* istanbul ignore next */
+    return null;
+  }
 
-        return {
-          datasource: GithubTagsDatasource.id,
-          registryUrl: `${parsedUrl.protocol}//${parsedUrl.host}`,
-          packageName,
-        };
+  private static goImportHeader(
+    res: string,
+    goModule: string
+  ): DataSource | null {
+    const importMatch = regEx(
+      `<meta\\s+name="go-import"\\s+content="([^\\s]+)\\s+([^\\s]+)\\s+([^\\s]+)">`
+    ).exec(res);
+    if (importMatch) {
```

Comment:
I think switch this to negative:

```suggestion
    if (!importMatch) {
```

And then return early.

Reply:
didnt even notice that xD yes sure.